__tests__/app.test.js

```diff
@@ -104,4 +124,59 @@ describe('/api/articles/:article_id/comments', () => {
     .get('/api/articles/99/comments')
     .expect(404)
   })
+
+  test('Responds with an error when given endpoint doesn exist', () => {
+    return request(app)
+    .get('/api/articles/notAnId/comments')
+    .expect(400)
+  })
+})
+
+describe('POST /api/articles/:article_id/comments', () => {
+  test('Responds with the body posted by the client', () => {
+    return request(app)
+    .post('/api/articles/2/comments')
+    .send({
+      author: 'butter_bridge',
+      article_id: 2,
+      body: "sdoihod o h oe hgoie jo ej eij oiejrfoiej foejri"
+    })
+    .expect(201)
+    .then((response) => {
+      expect(response.body.comment).toMatchObject({
+        body: "sdoihod o h oe hgoie jo ej eij oiejrfoiej foejri",
+        votes: 0,
+        author: 'butter_bridge',
+        article_id: 2,
+      })
+    })
+  })
+
+  test('Responds with an error when given article doesn exist', () => {
+    return request(app)
+    .post('/api/articles/99/comments')
+    .send({
+      author: 'butter_bridge',
+      article_id: 2,
+      body: "sdoihod o h oe hgoie jo ej eij oiejrfoiej foejri"
+    })
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe('Article not found');
+    });
+  })
+
+  test('Responds with an error when given endpoint doesn exist', () => {
+    return request(app)
+    .post('/api/articles/notAnId/comments')
+    .send({
+      author: 'butter_bridge',
+      article_id: 2,
+      body: "sdoihod o h oe hgoie jo ej eij oiejrfoiej foejri"
+    })
+    .expect(400)
+    .then((response) => {
+      expect(response.body.msg).toBe('Invalid input');
+    });
+  })
```

Comment:
This error testing that you have so far is looking good! 
There are more errors that could be made by the client in their request on this endpoint:
- A client may make a request where the `body` key is missing 
- A client may make a request where the `author` does not exist in the users table

Because this query is utilising INSERT, the _FOREIGN KEY_ and _NOT NULL_ constraints on these columns will cause PSQL errors. This means you don't need any manual logic for a `Promise.reject' in your model, just check for the PSQL error code you get back